src/int64_util.js

```diff
@@ -26,66 +30,92 @@ var POW2_31 = Math.pow(2, 31);
+            return (negative ? '-' : '') + low;
+        }
+        // Make it exactly 11 with leading zeros.
+        low = ('00000000000' + String(low % POW10_11)).slice(-11);
+        return (negative ? '-' : '') + String(high) + low;
     }
-    var high2 = b[o + 1] + (b[o] << 8);
-    // Lesser 11 digits with exceeding values but is under 53 bits capacity.
-    var low = b[o + 7] + (b[o + 6] << 8) + (b[o + 5] << 16)
-        + b[o + 4] * POW2_24  // Bit shift renders 32th bit as sign, so use multiplication
-        + (b[o + 3] + (b[o + 2] << 8)) * POW2_32 + high2 * 74976710656;  // The literal is 2^48 % 10^11
-    // 12th digit and greater.
-    var high = Math.floor(low / POW10_11) + high2 * 2814;  // The literal is 2^48 / 10^11
-    // Make it exactly 11 with leading zeros.
-    low = ('00000000000' + String(low % POW10_11)).slice(-11);
-    return (negative ? '-' : '') + String(high) + low;
-  }
 };
 
-Int64Util.fromDecimalString = function(text) {
-  var negative = text.charAt(0) === '-';
-  if (text.length < (negative ? 17 : 16)) {
-    // The magnitude is smaller than 2^53.
-    return new Int64(+text);
-  } else if (text.length > (negative ? 20 : 19)) {
-    throw new RangeError('Too many digits for Int64: ' + text);
-  } else {
-    // Most significant (up to 5) digits
-    var high5 = +text.slice(negative ? 1 : 0, -15);
-    var low = +text.slice(-15) + high5 * 2764472320;  // The literal is 10^15 % 2^32
-    var high = Math.floor(low / POW2_32) + high5 * 232830;  // The literal is 10^15 / 2^&32
-    low = low % POW2_32;
-    if (high >= POW2_31 &&
-        !(negative && high == POW2_31 && low == 0)  // Allow minimum Int64
-       ) {
-      throw new RangeError('The magnitude is too large for Int64.');
+Int64Util.fromDecimalString = function (text)
+{
+    var negative = text.charAt(0) === '-';
+    if (text.length < (negative ? 17 : 16))
+    {
+        // The magnitude is smaller than 2^53.
+        return new Int64(+text);
+    }
+    else if (text.length > (negative ? 20 : 19))
+    {
+        throw new RangeError('Too many digits for Int64: ' + text);
     }
-    if (negative) {
-      // 2's complement
-      high = ~high;
-      if (low === 0) {
-        high = (high + 1) & 0xffffffff;
-      } else {
-        low = ~low + 1;
-      }
-      high = 0x80000000 | high;
+    else
+    {
+        // Most significant (up to 5) digits
+        var high5 = +text.slice(negative ? 1 : 0, -15);
+        var low = +text.slice(-15) + high5 * 2764472320; // The literal is 10^15 % 2^32
+        var high = Math.floor(low / POW2_32) + high5 * 232830; // The literal is 10^15 / 2^&32
+        low = low % POW2_32;
+        // The equality operators were changed to '===' from '==' when this file was copied from the Thrift project
+        if (high >= POW2_31 &&
+            !(negative && high === POW2_31 && low === 0) // Allow minimum Int64
+        )
+        {
+            throw new RangeError('The magnitude is too large for Int64.');
```

Comment:
initial lowercase for all exception messages (no period at the end)

Reply:
fixed